c2rust-analyze/src/main.rs

```diff
@@ -576,6 +564,35 @@ fn run(tcx: TyCtxt) {
     }
     eprintln!("reached fixpoint in {} iterations", loop_count);
 
+    // For testing, putting #[c2rust_analyze_test::fail_before_rewriting] on a function marks it as
+    // failed at this point.
+    for &ldid in &all_fn_ldids {
+        if !util::has_test_attr(tcx, ldid, TestAttr::FailBeforeRewriting) {
+            continue;
+        }
+        gacx.mark_fn_failed(
+            ldid.to_def_id(),
+            PanicDetail::new("explicit fail_before_rewriting for testing".to_owned()),
+        );
+    }
+
+    // Before generating rewrites, add the FIXED flag to the signatures of all functions that
+    // failed analysis.
+    for did in gacx.iter_fns_failed() {
+        let lsig = match gacx.fn_sigs.get(&did) {
+            Some(x) => x,
+            None => continue,
+        };
```

Comment:
```suggestion
        let lsig = gacx.fn_sigs[&did];
```

This lookup shouldn't fail, right?

Reply:
Good idea.  In the future we might expand analysis to static initializers, and it would be good for this to panic if a non-function has failed analysis instead of silently ignoring it.